### PR TITLE
Rewrite pricing calculator

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
Pricing calculator had tons of issue so we decided to rewrite it in a better way

### Changes
Rewrite pricing calculator

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2291
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2293
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1899
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2292

image

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/router/index.ts [hunk 1]
```diff
@@ -469,7 +469,7 @@ function createDeployRoutes(): RouteRecordRaw[] {
       children: [
         {
           path: DashboardRoutes.Deploy.PricingCalculator,
-          component: () => import("../calculator/resource_pricing.vue"),
+          component: () => import("../calculator/PricingCalculator.vue"),
```
Comment (AlaaElattar) [inline packages/playground/src/router/index.ts:472]: i don't think we rename files with camelcase

**Comment by AlaaElattar** [comment]
- I built and ran `yarn install`, but the page not behaving right.
- The prices below shouldn't disappear before changing also the colors not changing.
- The spaces between toggles.

[Screencast from 03-05-2024 04:40:16 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/b888a0b7-0489-42a4-950e-d619195b5690)

File: packages/playground/src/router/index.ts (hunk 1 above)
Comment (MohamedElmdary) [reply packages/playground/src/router/index.ts]: Updated

**Comment by MohamedElmdary** [comment]
> * I built and ran `yarn install`, but the page not behaving right.
> * The prices below shouldn't disappear before changing also the colors not changing.
> * The spaces between toggles.
> 
>  Screencast.from.03-05-2024.04.40.16.PM.webm

update loading behaviour but what about toggles spacing?

**Comment by AlaaElattar** [comment]
- By default if user logged in, the balance toggle should be enabled

**Comment by AlaaElattar** [comment]
- The color not changing at all.
 
image

**Comment by MohamedElmdary** [comment]
> * The color not changing at all.
> 
> image

image
image

**Comment by AlaaElattar** [comment]
- My balance should my valid even if it's low. what do u think ?

image

File: packages/playground/src/calculator/pricing_calculator.vue [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                  Nodes allow several users to host various workloads on a single node
+                </p>
+              </div>
+            </VCol>
+          </VRow>
+
+          <VRow v-else>
+            <VCol cols="12">
+              <VAlert type="error">
+                Please provide a valid data inorder to calculate your deployment price correctly.
```
Comment (maayarosama) [inline packages/playground/src/calculator/pricing_calculator.vue:176]: I think the alert should be more clear, it should mention which field is invalid. what do you think?

File: packages/playground/src/calculator/pricing_calculator.vue [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                </p>
+              </div>
+            </VCol>
+            <VCol cols="6">
+              <div
+                class="rounded pa-4 h-100 d-flex justify-center align-center"
+                :style="{ background: computePackageColor(priceTask.data?.sharedPackage.package) }"
+              >
+                <p>
+                  Cost of reservation on a <strong v-text="'Shared Node'" /><br />
```
Comment (maayarosama) [inline packages/playground/src/calculator/pricing_calculator.vue:164]: I'm not sure if we should mention the package's name here or if the background color is enough?

File: packages/playground/src/calculator/pricing_calculator.vue (hunk 2 above)
Comment (MohamedElmdary) [reply packages/playground/src/calculator/pricing_calculator.vue:176]: there is already an error shown on that specific field ... we can know instantly which field has error without checking every field over and over

File: packages/playground/src/calculator/pricing_calculator.vue (hunk 3 above)
Comment (MohamedElmdary) [reply packages/playground/src/calculator/pricing_calculator.vue:164]: @maayarosama what do u mean? 
please add some ss if something is not working correctly

Comment (maayarosama) [reply packages/playground/src/calculator/pricing_calculator.vue:164]: Let's ignore my suggestion here as it's not needed

**Review by AlaaElattar — APPROVED** [review]
Great Job

**Review by maayarosama — APPROVED** [review]

**Pull request merged**